vowpalwabbit/parse_args.cc

```diff
@@ -206,8 +205,8 @@ void parse_dictionary_argument(vw& all, std::string str)
                                                          << ", opening failed");
   }
 
-  feature_dict* map = &calloc_or_throw<feature_dict>();
-  map->init(1023, nullptr, substring_equal);
+  std::shared_ptr<feature_dict> map = std::make_shared<feature_dict>();
```

Comment:
```suggestion
  auto map = std::make_shared<feature_dict>();
```

Reply:
for this, I think an explicit definition is better for readability